framework/src/util/Utils.cpp

```diff
@@ -90,8 +110,28 @@ std::string library_suffix()
 
 }
 
+
 namespace cppmicroservices {
+  //-------------------------------------------------------------------
+  // Unicode Utility functions
+  //-------------------------------------------------------------------
+#ifdef US_PLATFORM_WINDOWS
+  std::wstring ToWString(const std::string& inStr)
+  {
+    int wchars_count = MultiByteToWideChar(CP_UTF8, 0, inStr.c_str(), -1, NULL, 0);
+    std::unique_ptr<wchar_t[]> wBuf(new wchar_t[wchars_count]);
+    MultiByteToWideChar(CP_UTF8, 0, inStr.c_str(), -1, wBuf.get(), wchars_count);
+    return wBuf.get();
```

Comment:
The returned object is copied before the std::unique_ptr is destroyed.
https://stackoverflow.com/questions/275214/scope-and-return-values-in-c

Thanks for the vector suggestion! :-)